Action doesn't work on ubuntu-24.04 / ubuntu-latest (when it's ubuntu-24.04) because `sbt` is missing

As noted in:
* https://github.com/actions/setup-java/issues/712

GitHub's ubuntu-24.04 runners don't include `sbt`.

This means that a workflow like this (in a repository `github-organization-owner/github-repository` in a file `.github/workflows/scala-steward.yml`):
```yaml
on:
  workflow_dispatch:
  schedule:
    - cron: "30 12 * * MON"

name: Scala Steward
permissions:
  contents: write
  pull-requests: write

jobs:
  scala-steward:
    runs-on: ubuntu-latest
    name: Scala Steward
    steps:
      - name: Scala Steward
        uses: scala-steward-org/scala-steward-action@v2
        with:
          author-email: ${{ env.email }}
          author-name: ${{ env.name }}
          scala-steward-version: 0.30.2
```

... will result in this error:

```java
java.io.IOException: Cannot run program "sbt" (in directory "/home/runner/scala-steward/workspace/repos/github-organization-owner/github-repository"): error=2, No such file or directory
  	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1143)
  	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
  	at org.scalasteward.core.io.process$.$anonfun$createProcess$2(process.scala:106)
  	at flatMap @ org.scalasteward.core.io.process$.createProcess(process.scala:104)
  	at flatMap @ org.scalasteward.core.io.process$.slurp(process.scala:62)
  	at delay @ org.typelevel.log4cats.slf4j.internal.Slf4jLoggerInternal$Slf4jLogger.isDebugEnabled(Slf4jLoggerInternal.scala:73)
  	at ifM$extension @ org.typelevel.log4cats.slf4j.internal.Slf4jLoggerInternal$Slf4jLogger.info(Slf4jLoggerInternal.scala:99)
  	at >>$extension @ org.scalasteward.core.update.artifact.ArtifactMigrationsLoader.loadMigrations(ArtifactMigrationsLoader.scala:45)
  	at main$ @ org.scalasteward.core.Main$.main(Main.scala:23)
  	at main$ @ org.scalasteward.core.Main$.main(Main.scala:23)
  Caused by: java.io.IOException: error=2, No such file or directory
  	at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
  	at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:314)
  	at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:244)
  	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1110)
  	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1073)
  	at org.scalasteward.core.io.process$.$anonfun$createProcess$2(process.scala:106)
  	at cats.effect.IOFiber.runLoop(IOFiber.scala:1004)
  	at cats.effect.IOFiber.asyncContinueSuccessfulR(IOFiber.scala:1368)
  	at cats.effect.IOFiber.run(IOFiber.scala:113)
  	at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:702)
```

The fix is to conditionally install `sbt`.


There are various ways to do this. Sadly, I can't find any particularly _good_ ways to do this.

* https://github.com/actions/toolkit/issues/1458 / https://github.com/actions/toolkit/issues/1510 mean that a node action isn't in a position to call a composite action
* a composite action can't use a local action with `pre` steps which means things would be a bit messy if one tried to rearrange things (but I'm going to try because it's interesting)
* https://github.com/sbt/setup-sbt is a composite action that can set up `sbt` (and does fancy things for caching) but see the previous two points
* one could write code to reimplement `setup-sbt` or manually try the various poor ways of installing `sbt` by hand
* this action should at least have better error handling for when `sbt` is missing as the current top level output is really unhelpful:
    ```
    Run scala-steward-org/scala-steward-action@v2
    ✓ Scala Steward workspace created
    Trying to restore workspace contents from cache...
    Launching org.scala-steward:scala-steward-core_2.13:0.30.2
    Saving workspace to cache...
    Error:  ✕ Launching org.scala-steward:scala-steward-core_2.13:0.30.2 failed
    ```

--- a/README.md
+++ b/README.md
@@ -37,8 +37,6 @@ jobs:
     runs-on: ubuntu-latest
     name: Scala Steward
     steps:
-      - name: Install sbt
-        uses: sbt/setup-sbt@v1
       - name: Scala Steward
         uses: scala-steward-org/scala-steward-action@v2
 ```
@@ -134,8 +132,6 @@ jobs:
     runs-on: ubuntu-latest
     name: Launch Scala Steward
     steps:
-      - name: Install sbt
-        uses: sbt/setup-sbt@v1
       - name: Launch Scala Steward
         uses: scala-steward-org/scala-steward-action@v2
         with:
@@ -333,8 +329,6 @@ You can manually trigger workflow runs using the [workflow_dispatch](https://doc
      runs-on: ubuntu-latest
      name: Launch Scala Steward
      steps:
-       - name: Install sbt
-         uses: sbt/setup-sbt@v1
        - name: Launch Scala Steward
          uses: scala-steward-org/scala-steward-action@v2
          with:
@@ -392,8 +386,6 @@ When using the `github-app-*` inputs, Scala Steward will always retrieve the lis
      runs-on: ubuntu-latest
      name: Launch Scala Steward
      steps:
-       - name: Install sbt
-         uses: sbt/setup-sbt@v1
        - name: Launch Scala Steward
          uses: scala-steward-org/scala-steward-action@v2
          with:

--- a/src/modules/coursier.ts
+++ b/src/modules/coursier.ts
@@ -10,7 +10,7 @@ import {type NonEmptyString} from '../core/types'
  * Installs `coursier` and add its executable to the `PATH`.
  *
  * Once coursier is installed, installs `scalafmt`
- * `scalafix` and `scala-cli` tools.
+ * `scalafix`, `sbt` and `scala-cli` tools.
  *
  * Throws error if the installation fails.
  */
@@ -32,7 +32,7 @@ export async function install(): Promise<void> {
 
     await exec.exec(
       'cs',
-      ['install', 'scalafmt', 'scalafix', 'scala-cli', '--install-dir', binary],
+      ['install', 'scalafmt', 'scalafix', 'scala-cli', 'sbt', '--install-dir', binary],
       {
         silent: true,
         listeners: {stdline: core.debug, errline: core.debug},
@@ -50,6 +50,8 @@ export async function install(): Promise<void> {
     const scalafixVersion = await execute('cs', 'launch', 'scalafix', '--', '--version')
 
     core.info(`✓ Scalafix installed, version: ${scalafixVersion.trim()}`)
+
+    core.info('✓ SBT installed')
 
     core.info('✓ scala-cli installed')
   } catch (error: unknown) {

